pvanalytics/system.py

```diff
@@ -1,10 +1,280 @@
+    or fixed. If performed, this result must be consistent with the fit
+    to the upper envelope. If not, :py:const:`Tracker.UNKNOWN`
+    is returned.
+
+    Parameters
+    ----------
+    series : Series
+        Timezone localized Series of power or irradiance data.
+    daytime : Series
+        Boolean Series with True for times that are during the day.
+    clipping : Series
+        Boolean Series identifying where power or irradiance is being
+        clipped.
+    clip_max : float, default 0.1
+        If the fraction of data flagged as clipped is greater than
+        `clip_max` then it cannot be determined whether the system is
+        tracking or fixed and :py:const:`Tracker.UNKNOWN` is returned.
+    envelope_quantile : float, default 0.995
+        Quantile used to determine the upper power or irradiance
+        envelope.
+    envelope_min_fraction : float, default 0.05
+        After calculating the power or irradiance envelope, data less
+        than `envelope_min_fraction` times the maximum of the envelope
+        is removed. This excludes data from morning and evening that
+        may interfere with curve fitting.
+    fit_median : boolean, default True
+        Perform a secondary fit with the median power or irradiance to
+        validate that the profile is consistent through the entire
+        data set.
+    median_min_fraction : float, default 0.025
+        After calculating the median power or irradiance at each
+        minute, data less than `median_min_fraction` times the maximum
+        is removed. This excludes data from morning and evening that
+        may interfere with curve fitting.
+    median_r2_min : float, default 0.9
+        Minimum :math:`r^2` for a curve fit to the median power or
+        irradiance at each minute of the day (Applies only if
+        `fit_median` is True).
+    fit_params : dict or None, default None
+        Minimum r-squared for curve fits according to the fraction of
+        data with clipping. This should be a dictionary with tuple
+        keys and dictionary values. The key must be a 2-tuple of
+        ``(clipping_min, clipping_max)`` where the values specify the
+        minimum and maximum fraction of data with clipping for which
+        the associated fit parameters are applicable. The values of
+        the dictionary are themselves dictionaries with keys
+        ``'fixed'`` and ``'tracking'``, which give the minimum
+        :math:`r^2` for the curve fits, and ``'fixed_max'`` which
+        gives the maximum :math:`r^2` for a quadratic fit if the
+        system appears to have a tracker.
+
+        If None :py:data:`PVFLEETS_FIT_PARAMS` is used.
+    seasonal_split : tuple of list-like, default ((5,6,7,8), (11,12,1,2))
+        Two-tuple specifying a set of winter months and a set of summer
+        months. The order is not important. The months are specified
+        as integers between 1 and 12. The default value works well for North
```

Comment:
```suggestion
    seasonal_split : tuple of list-like, default ((5,6,7,8), (11,12,1,2))
        Two-tuple specifying tuples of months for summer and winter seasons.
        The months are specified
        as integers between 1 and 12. The default value works well for North
```

I hate to churn on this but I think it's still less clear than it should be. "The order is not important" true in that the code doesn't care which tuple is considered summer, but I think it would help users to identify one element as summer and the other as winter.

Pushing that a bit further, what if the default was a dict instead? `{'summer': (5, 6, 7, 8), 'winter': (11, 12, 1, 2)}`

Reply:
I totally agree. I was just looking at that language. You make a good point about being "explicit" about what is winter and what is summer. Rather than a dictionary, what about two parameters? `winter=(11, 12, 1, 2)` and `summer = (5, 6, 7, 8)` ?

Might want to change the semantics a bit too, and allow `None` for either or both seasons. I will need to write some additional tests in that case, but the more I think about it the more it seems like the right thing to do.

Reply:
I'm OK with two parameters, but I think it should be easy to disable this optional check which would be an advantage of a single argument (pass `None` to skip)